### PR TITLE
Handle argument "facecolors=None" correctly in plot_surface()

### PR DESCRIPTION
**Pull request by ocastany**
## PR Summary

Handle `plot_surface(facecolors=None)` correctly. 
This patch avoids the error: `UnboundLocalError: local variable 'color' referenced before assignment`.

## PR Checklist

**Documentation and Tests**
- [N/A] Has pytest style unit tests (and `pytest` passes)
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] New plotting related features are documented with examples.

**Release Notes**
- [N/A] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [N/A] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [N/A] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Comment by rcomer** [comment]
Thank you for your contribution @ocastany.  Could you provide an example that demonstrates the problem you are fixing?

**Comment by ocastany** [comment]
Here is an example showing the problem before the patch:
```
> from pylab import *
> fig = figure()
> ax = fig.add_subplot(projection='3d')
> X,Y = meshgrid(arange(5), arange(5))
> Z = sqrt(X**2 + Y**2)
> ax.plot_surface(X,Y,Z, facecolors=None)

UnboundLocalError: local variable 'color' referenced before assignment
```

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 1]
```diff
@@ -1599,11 +1599,13 @@ def plot_surface(self, X, Y, Z, *, norm=None, vmin=None,
         if 'facecolors' in kwargs:
             fcolors = kwargs.pop('facecolors')
         else:
+            fcolors = None
```
Comment (greglucas) [inline lib/mpl_toolkits/mplot3d/axes3d.py:1602]:
```suggestion
        fcolors = kwargs.pop('facecolors', None)
```

**Review by greglucas — COMMENTED** [review]
Could you add your example as a test so this doesn't come back again in the future?

**Comment by ocastany** [comment]
Sorry, I don't know where and how to "add my example as a test".

**Comment by melissawm** [comment]
Hi @ocastany - you can look at the [test_axes3d.py](https://github.com/matplotlib/matplotlib/blob/main/lib/mpl_toolkits/mplot3d/tests/test_axes3d.py) file and add your example as a test that needs to pass. You can also look at https://matplotlib.org/stable/devel/testing.html for more details on writing and verifying tests for Matplotlib. Cheers!

**Comment by ocastany** [comment]
I added a test function in my GitHub repository. Can you use it?

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py [hunk 2]
```diff
@@ -591,6 +591,14 @@ def test_surface3d_masked():
     ax.view_init(30, -80, 0)
 
 
+def test_plot_surface_None_arg():
+    fig = plt.figure()
+    ax = fig.add_subplot(projection='3d')
+    x, y = np.meshgrid(np.arange(5), np.arange(5))
+    z = x + y
+    ax.plot_surface(x,y,z, facecolors=None)
```
Comment (greglucas) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:599]:
```suggestion
@check_figures_equal(extensions=["png"])
def test_plot_surface_None_arg(fig_test, fig_ref):
    x, y = np.meshgrid(np.arange(5), np.arange(5))
    z = x + y
    
    ax_test = fig_test.add_subplot(projection='3d')
    ax_test.plot_surface(x, y, z, facecolors=None)
    
    ax_ref = fig_ref.add_subplot(projection='3d')
    ax_ref.plot_surface(x, y, z)
```

**Review by greglucas — COMMENTED** [review]
Yep, I can see your test and it looks good! Just a minor suggestion that we can do an image comparison that the `facecolors=None` explicit argument should do the same thing as the default. We have a test decorator to help with that which I've added as a suggestion.

**Comment by ocastany** [comment]
This looks good to me

**Comment by greglucas** [comment]
pre-commit.ci autofix

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py [hunk 3]
```diff
@@ -591,6 +591,18 @@ def test_surface3d_masked():
     ax.view_init(30, -80, 0)
 
 
+@check_figures_equal(extensions=["png"])
+def test_plot_surface_None_arg(fig_test, fig_ref):
+    x, y = np.meshgrid(np.arange(5), np.arange(5))
+    z = x + y
+
```
Comment (greglucas) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:598]:
```suggestion

```

File: lib/mpl_toolkits/mplot3d/tests/test_axes3d.py [hunk 4]
```diff
@@ -591,6 +591,18 @@ def test_surface3d_masked():
...
 
 
+@check_figures_equal(extensions=["png"])
+def test_plot_surface_None_arg(fig_test, fig_ref):
+    x, y = np.meshgrid(np.arange(5), np.arange(5))
+    z = x + y
+    
+    ax_test = fig_test.add_subplot(projection='3d')
+    ax_test.plot_surface(x, y, z, facecolors=None)
+
```
Comment (greglucas) [inline lib/mpl_toolkits/mplot3d/tests/test_axes3d.py:601]:
```suggestion

```

**Comment by greglucas** [comment]
Sorry @ocastany, it looks like the suggestions I had added additional white-space on some lines. precommit.ci can't autofix these because the PR is from your protected branch, so I suggested some additional changes to remove that which you will have to commit to your own branch.

**Review by rcomer — APPROVED** [review]

**Review by greglucas — APPROVED** [review]

**Comment by greglucas** [comment]
Thanks, @ocastany!

**Pull request merged**